transactId-library-ios/transactId-library-ios/Bip75/Bip75ServiceNetki.swift

```diff
@@ -264,7 +248,135 @@ class Bip75ServiceNetki: Bip75Service {
+                            
+                            if (!isCertificateOwnerChainValid) {
+                                throw Exception.InvalidCertificateChainException(ExceptionMessages.certificateValidationInvalidOwnerCertificateCA)
+                            }
+                            
+                            let isSignatureValid = try messageAttestation.validateMessageSignature(requireSignature: messageOriginator.isPrimaryForTransaction)
+                            
+                            if (!isSignatureValid) {
+                                throw Exception.InvalidSignatureException(ExceptionMessages.signatureValidationInvalidOwnerSignature)
+                            }
+                        }
+                    }
+                    
+                    try messagePayment.beneficiaries.forEach { (messageBeneficiary) in
+                        try messageBeneficiary.attestations.forEach { (messageAttestation) in
+                            let isCertificateOwnerChainValid = try self.validateCertificate(pkiType: messageAttestation.getMessagePkiType(), certificate: messageAttestation.pkiData.toString())
+                            if (!isCertificateOwnerChainValid) {
+                                throw Exception.InvalidCertificateChainException(ExceptionMessages.certificateValidationInvalidOwnerCertificateCA)
+                            }
+                        }
+                    }
+                    return true
+                }
+            }
+        }
+        return false
+    }
+    
+    func parsePayment(paymentBinary: Data, recipientParameters: RecipientParameters?) throws -> Payment? {
+        if let protocolMessageMetadata = try paymentBinary.extractProtocolMessageMetadata() {
+            if let messagePaymentData = try paymentBinary.getSerializedMessage(encrypted: protocolMessageMetadata.encrypted, recipientParameters: recipientParameters) {
+                if let messagePayment = try messagePaymentData.toMessagePayment() {
+                    return try messagePayment.toPayment(protocolMessageMetadata: protocolMessageMetadata)
+                }
+            }
+        }
+        return nil
+    }
+    
+    //MARK: PaymentACK
+
+    func createPaymentACK(paymentAckParameters: PaymentAckParameters) throws -> Data? {
+        if let messagePaymentACK = paymentAckParameters.payment?.toMessagePaymentACK(memo: paymentAckParameters.memo) {
+            let paymentACK = try messagePaymentACK.serializedData()
+            return try paymentACK.toProtocolMessageData(messageType: .paymentACK,
+                                                        messageInformation: paymentAckParameters.messageInformation,
+                                                        senderParameters: paymentAckParameters.senderParameters,
+                                                        recipientParameters: paymentAckParameters.recipientParameters)
+        }
+        return nil
```

Comment:
Not sure how swift works.
Should we return an exception saying why we could not create the PaymentACk instead of a nil?

Reply:
This code will never be executed if we have some exception. We have to return nil here because we have to return something.